server/compactor/src/main/java/org/apache/accumulo/compactor/Compactor.java

```diff
@@ -184,7 +184,7 @@ protected void startCancelChecker(ScheduledThreadPoolExecutor schedExecutor,
         TimeUnit.MILLISECONDS);
   }
 
-  protected void checkIfCanceled() {
+  protected synchronized void checkIfCanceled() {
```

Comment:
What does the addition of synchonized here defend against?   There are reads from metadata table and ZK in this method, so the sync could possibly cause problems.

Reply:
it's being called by a background thread every 5 seconds and from the main thread when the compaction is completed. I was just trying to serialize the calls.

Reply:
I noticed the `cancel()` method call on `CompactionJobHolder` is synchronized, so maybe that is enough. Could add another check to this method to see if the compactionThread is already interrupted so we don't keep calling interrupt while waiting for the thread to die.

Reply:
> I noticed the cancel() method call on CompactionJobHolder is synchronized, so maybe that is enough.

I think the existing sync in CompactionJobHolder will prevent any race conditions from happening.

Reply:
> Could add another check to this method to see if the compactionThread is already interrupted so we don't keep calling interrupt while waiting for the thread to die.

It may be useful to keep interrupting a thread because some code may ignore interrupts w/o throwing an exception.

Reply:
I have removed the synchronized modifier from this method.